vizzys fgd update (except icons)

meatspace opened the pull request:
This is based on the "mapbase" branch of the mapbase-source/source-fgds repo

I did not add custom Icons/entity model helpers in this specific commit. 

I did not include the custom soundscapes lines I added. 

"Touch Opens" flag for the base door class was changed from 1 to 0, not sure if that's intentional on your part but imo its better off by default.

Whats new?

many missing inputs/outputs
many improved entity descriptions
many entity helper lines/spheres
many typo fixes

meatspace committed:
vizzys fgd (except icons)

This is every change my updated .fgd makes aside from a few key differences:

This is based on the "mapbase" branch of the mapbase-source/source-fgds repo, which I have changed to be the main branch (not the super outdated main branch.)

I did not add custom Icons/entity model helpers in this specific commit. They will be in the next commit. This is because I'm unsure if you wish to include these.

I did not include the custom soundscapes lines I added. They will be in the next commit. This is because I'm unsure if you wish to include these.

"Touch Opens" flag for the base door class was changed from 1 to 0, not sure is that's intentional on your part but imo its better off by default.

as for the changelog and icons, consider downloading them here:
https://gamebanana.com/tools/14943
(however its slightly outdated compared to this commit)

Whats new?

many missing inputs/outputs
many improved entity descriptions
many entity helper lines/spheres
many typo fixes

meatspace committed:
oops

forgot this change

meatspace committed:
more changes

meatspace committed:
typo

well, missing space really

meatspace committed:
even more changes

wow

meatspace committed:
update to mapbase 7.3 beta changes

BLIXI! USE MY FGD UPDATE AND MY LIFE IS YOURS!

Blixibon commented on base.fgd:
```diff
@@ -3233,6 +3261,43 @@
 	output PlayersOutCount(integer) : "Fired after a count, and contains the number of players found outside the zone."
 ]
 
+// start commentary section added by vizzys
```
Why add the commentary entities?

meatspace replied:
i figured someone would use it, but I dont have the information you have about the missing other modes (image, etc) and I notice now I forgot to remove my custom icon here so Ill get rid of it.

Blixibon commented on base.fgd:
```diff
@@ -4949,6 +5027,7 @@
 	input SetFModRate(float) : "Set the modulation cycles per second from -10.0f to 10.0."
 	input SetFModAmplitude(float) : "Set the modulation extents from 0.0f to 10.0."
 	input RandomizeFMod(float) : "Randomize the frequency modulation by an extremeness of 0.0 to 1.0."
+	input GetFMod(float) : "Outputs the current modulation settings in console."
```
```suggestion
	input GetFMod(void) : "Outputs the current modulation settings in console."
```

Blixibon commented on halflife2.fgd:
```diff
@@ -925,25 +929,25 @@
+		"weapon_357" : ".357 Magnum (Revolver)"
 		"weapon_ar2" : "AR2"
-		"weapon_stunstick" : "Stun Stick"
+		"weapon_citizenpackage" : "Citizen Package  (Ration)"
```
```suggestion
		"weapon_citizenpackage" : "Citizen Package (Ration)"
```

Blixibon commented on halflife2.fgd:
```diff
@@ -3684,10 +3705,10 @@
+		1 : "Downtrodden (Clean, blue uniform)"
+		2 : "Refugee (Dirty, ragged clothing)"
+		3 : "Rebel (Full resistance getup and combat gear)"
+		4 : "Unique (Set a particular model with Model)"
```
Different description compared to npc_citizen above

meatspace replied:
will fix

Blixibon commented on halflife2.fgd:
```diff
@@ -5409,6 +5434,11 @@
 	"A way to access the original HL2 jeep in Episodic. Just use prop_vehicle_jeep if you're using the original HL2 version."
 [
 	model(studio) : "World model" : "models/buggy.mdl"
+	EnableGun(choices) : "Has Gun" : 0 : "Whether the jeep's tau cannon is enabled or disabled." =
+	[
+		0 : "No"
+		1 : "Yes"
+	]
```
Why is this 0 by default?

meatspace replied:
I just assumed off would be the more likely choice when manually placing jeep/buggy in hammer. Ill change it to 1 I suppose

Blixibon reviewed (changes requested):
I was a bit confused to see some of Mapbase v7.3's FGD changes already in here. For future reference, I was planning on committing those myself later. Adding them like this ahead of time can lead to merge conflicts due to them being added at the same time through separate commits.

I have some questions about the content in this branch, although I would be interested in integrating this for v8.0 with some additional changes.

meatspace committed:
actual mapbase 7.3 update

for real this time using blixis fixed fgds

meatspace committed:
Revert "actual mapbase 7.3 update"

This reverts commit c72fc858b83282a61c77ca0bcdf592b91282703c.

meatspace committed:
Revert "update to mapbase 7.3 beta changes"

This reverts commit 844dae11a2c873b933ac7bc07386f286fc7e3624.

meatspace committed:
change GetFMod to (void)

Co-authored-by: Blixibon <acj3017@gmail.com>

meatspace committed:
remove space between Package (Ration)

Co-authored-by: Blixibon <acj3017@gmail.com>

meatspace committed:
the requested changes

meatspace commented:
I think I fixed it, feel free to add more comments and such. Anything can be edited to your hearts desire ofc, especially descriptions which some are worded in a terse way to not exceed the 128 character string limitation

meatspace committed:
some BaseTrain stuff I forgot

 apparently

meatspace committed:
Merge remote-tracking branch 'upstream/mapbase' into mapbase

meatspace committed:
Revert "Merge remote-tracking branch 'upstream/mapbase' into mapbase"

This reverts commit 0676a22621b15dd5781e7dc2edf35ddba26fe8a3, reversing
changes made to d83a06ea20134eec8342e9827f264ae448177b6f.

meatspace committed:
7.3 stuff. i hate github

meatspace commented:
merge conflicts suck u were right... iam bad at computer
if you cant salvage this just let me know, ill resubmit a new PR instead.

anyway, if you want this to be some big update thing I think including the custom icons I made would be good

I keep that all in a seperate repo (havent update it for 7.3 yet)
[(https://github.com/meatspace/vizzys_fgds)](https://github.com/meatspace/vizzys_fgds)

meatspace committed:
editing .fgds at 3am bad

you forget stuff

meatspace committed:
small consistency fixes

meatspace committed:
the gravity gun update

you can call it the zero field point manipulator... if you want.

meatspace committed:
eagle eyed TeamSpen210

meatspace committed:
env_steam tweaks

from another shameless peek at TeamSpen210s cool fgd

meatspace committed:
gender neutrality pass

because player character isnt always male. do vorts lay eggs?

meatspace committed:
more neutrality

If I don't survive, tell my wife hello

meatspace committed:
neutrality and clarity

meatspace committed:
misc stuff + helpers

meatspace committed:
more helpers

havent tested tbh but look gud

meatspace committed:
smol fix

meatspace committed:
grammar, helpers, parentable physic

energybeam kind of suggest beam already so i removed +beams in parentheses. sprite helper is from mrfunreals l4d2 fgd. physics props can be parented afaik

meatspace committed:
parity with my other repo

some typos and grammar fixed

meatspace committed:
some inputs/outputs, combine lock door

meatspace committed:
world space glow sprite override, and other details/helpers

meatspace committed:
some consistency fixes

meatspace committed:
stuff + chained door helper

should b ok

meatspace committed:
mo inputs, kvs, lines, types, typos fixd

disabled move_track cause it doesnt exist. based on teamspen fgd mostly. input names validated with hl2 datamap dump on vdc

meatspace committed:
magnet input and apc helper

thas all

meatspace committed:
autocannon entity clarification

meatspace committed:
helper lines for days

and a few missing inputs

meatspace committed:
classes, helpers, etc

meatspace committed:
two smol changes

meatspace committed:
added missing parent fields

to env_beam and info_camera_link
